parameter/FloatingPointParameterType.cpp

```diff
@@ -0,0 +1,349 @@
+        }
+    } else {
+        // Provide value space only if not the default one
+        if (configurationAccessContext.valueSpaceIsRaw()) {
+
+            xmlConfigurableElementSettingsElement.setAttribute("ValueSpace", "Raw");
+        }
+    }
+}
+
+bool CFloatingPointParameterType::fromXml(const CXmlElement& xmlElement,
+                                          CXmlSerializingContext& serializingContext)
+{
+    // Size. The XSD fixes it to 32
+    uint32_t uiSizeInBits = 32;
+    xmlElement.getAttribute("Size", uiSizeInBits);
+
+    // Size support check
+    if (uiSizeInBits != sizeof(float) * CHAR_BIT) {
+
+        serializingContext.setError("Unsupported size (" + CUtility::toString(uiSizeInBits) +
+            ") for " + getKind() + " " + xmlElement.getPath() + ". For now, only 32 is supported.");
+
+        return false;
+    }
+
+    // Set the size
+    setSize(uiSizeInBits / CHAR_BIT);
+
+    // Bounds
+    xmlElement.getAttribute("Min", _fMin);
+    xmlElement.getAttribute("Max", _fMax);
+
+    if (_fMin > _fMax) {
+        serializingContext.setError("Min (" + CUtility::toString(_fMin) +
+                ") can't be greater than Max (" + CUtility::toString(_fMax) + ")");
+        return false;
+    }
+
+    return base::fromXml(xmlElement, serializingContext);
+}
+
+bool CFloatingPointParameterType::toBlackboard(
+        const string& strValue,
+        uint32_t& uiValue,
+        CParameterAccessContext& parameterAccessContext) const
+{
+    // Check Value integrity
+    if (CUtility::isHexadecimal(strValue) && !parameterAccessContext.valueSpaceIsRaw()) {
+
+        parameterAccessContext.setError("Hexadecimal values are not supported for "
+                                        + getKind()
+                                        + " when selected value space is real: "
+                                        + strValue);
+
+        return false;
+    }
+
+    if (parameterAccessContext.valueSpaceIsRaw()) {
+
+        uint32_t uiData;
+
+        // Intepret data
+        if (!convertTo(strValue, uiData)) {
+
+            parameterAccessContext.setError("Value " + strValue + " is invalid");
+
+            return false;
+        }
+
+        // Reinterpret as float
+        float fData = reinterpret_cast<const float&>(uiData);
+
+        // Check against NaN
+        if (fData != fData) {
+
+            parameterAccessContext.setError("Value " + strValue + " is not a number");
+
+            return false;
+        }
+
+        // Check against range
+        if (!checkValueAgainstRange(fData)) {
+
+            setOutOfRangeError(strValue, parameterAccessContext);
+
+            return false;
+        }
+
+        uiValue = uiData;
+
+        return true;
+    }
+    else {
+
+        float fValue;
+
+        // Intepret data
+        if (!convertTo(strValue, fValue)) {
+
+            parameterAccessContext.setError("Value " + strValue + " is invalid");
+
+            return false;
+        }
+
```

Comment:
no need: in this case, convertTo already takes care of that. In the previous case, we are converting the user input to a uint32 as if it was the memory's content. The memory content could represent NaN or an infinite value, hence the need to check.